apps/meteor/client/views/room/components/body/composer/messageBox/hooks/useMarkdownPreview.ts

```diff
@@ -0,0 +1,45 @@
+import type { IRoom } from '@rocket.chat/core-typings';
+import type { Root } from '@rocket.chat/message-parser';
+import { useMethod } from '@rocket.chat/ui-contexts';
+import type { SubscriptionWithRoom } from '@rocket.chat/ui-contexts';
+import { useState } from 'react';
+
+import { getChannel, getChannelMentions, getUser, getUserMentions, textToMessageToken } from '../utils/messageMarkdownUtils';
+import type { UserMention } from '../utils/messageMarkdownUtils';
+
+export const useMarkdownPreview = (rid: IRoom['_id']) => {
```

Comment:
Custom hook for managing and displaying previews of markdown-formatted messages.